### DISCUSSION TITLE
Generating a *.target

### DISCUSSION THREAD

**Discussion by merks** [eclipse-tycho/tycho#1903] [Q&A] [answered]
For this issue:

https://github.com/eclipse-cbi/targetplatform-dsl/issues/124

I am investigating how one could generate a *.target from a *.tdp on the fly as part of a Tycho build.  But the target platform resolution always starts very early and I've not figured out how to generate the *.target before Tycho tries to resolve it, which seems to be always the first thing it tries to do:
```
[Console output redirected to file:D:\Users\merks\cbi-tpd\git\targetplatform-dsl\build.log]
[INFO] Scanning for projects...
[ERROR] Setting target failed: target definition file 'D:\Users\merks\cbi-tpd\git\targetplatform-dsl\target-platforms\default.target' not found in project 'org.eclipse.cbi.targetplatform'. -> [Help 1]
```

Looking here:

https://tycho.eclipseprojects.io/doc/latest/target-platform-configuration/plugin-info.html

I see no mention of phases or the usual stuff.  So I'm starting to get the impression that what I'm trying to achieve, run something before the *.target is read, is not possible...

**Comment by laeubi** [comment, marked as answer]
This is not really possible with the "classic" resolver (but could work with the enhanced one), but it would actually be more valuable to not generated it manually but let Tycho do handle it nativly:
- https://github.com/eclipse-tycho/tycho/issues/32

The points one likely want to adjust are:
- `org.eclipse.tycho.targetplatform.TargetDefinitionFile.isTargetFile(File)` so it accepts files with `.tpd` and `.target` extension
- `org.eclipse.tycho.targetplatform.TargetDefinitionFile.read(File)` to first perform the conversion from tpd>target and then do a "normal" read or parse the tpd file in a way that produces a `TargetDefinitionFile` directly

The it should work to use either of both (probably there are some places that could still need adjustments like deploy of tpd file but this can be ignored as a first step, e.g. one maybe likes to have a deploy step that transform the tpd file and publish it in the target format).

**Reply by merks** [reply]
That's a very helpful answer.  Thanks.
